cwl/src/main/scala/cwl/TypeAliases.scala

```diff
@@ -30,6 +30,11 @@ trait TypeAliases {
       StepInputExpressionRequirement :+:
       CNil
 
+  type Hint =
+    Requirement :+:
```

Comment:
Having an explicit `Hint` type with a `Requirement` as an element of its `CoProduct` makes it easy to `select` out only `Requirement`s.